evm/src/cpu/kernel/asm/account_code.asm

```diff
@@ -0,0 +1,130 @@
+    // stack: offset, offset < code_length, offset, code_length, dest_offset, i, size, retdest
+    %mload_current(@SEGMENT_KERNEL_ACCOUNT_CODE)
+    // stack: opcode, offset < code_length, offset, code_length, dest_offset, i, size, retdest
+    %stack (opcode, offset_lt_code_length, offset, code_length, dest_offset, i, size, retdest) -> (offset_lt_code_length, 0, opcode, offset, code_length, dest_offset, i, size, retdest)
+    // If `offset >= code_length`, use `opcode=0`. Necessary since `SEGMENT_KERNEL_ACCOUNT_CODE` might be clobbered from previous calls.
+    %select_bool
+    // stack: opcode, offset, code_length, dest_offset, i, size, retdest
+    DUP4
+    // stack: dest_offset, opcode, offset, code_length, dest_offset, i, size, retdest
+    %mstore_main
+    // stack: offset, code_length, dest_offset, i, size, retdest
+    %increment
+    // stack: offset+1, code_length, dest_offset, i, size, retdest
+    SWAP2
+    // stack: dest_offset, code_length, offset+1, i, size, retdest
+    %increment
+    // stack: dest_offset+1, code_length, offset+1, i, size, retdest
+    SWAP3
+    // stack: i, code_length, offset+1, dest_offset+1, size, retdest
+    %increment
+    // stack: i+1, code_length, offset+1, dest_offset+1, size, retdest
+    %stack (i, code_length, offset, dest_offset, size, retdest) -> (i, size, code_length, offset, dest_offset, retdest)
+    %jump(extcodecopy_loop)
+
+extcodecopy_end:
+    %stack (i, size, code_length, offset, dest_offset, retdest) -> (retdest)
+    JUMP
+
+
+// Loads the code at `address` in the `SEGMENT_KERNEL_ACCOUNT_CODE` at the current context and starting at offset 0.
+// Checks that the hash of the loaded code corresponds to the `codehash` in the state trie.
+// Pre stack: address, retdest
+// Post stack: extcodesize(address)
+load_code:
+    // stack: address, retdest
+    %extcodehash
+    // stack: codehash, retdest
+    PROVER_INPUT(account_code::length)
+    // stack: code_length, codehash, retdest
+    PUSH 0
+
+// Loop non-deterministically querying `code[i]` and storing it in `SEGMENT_KERNEL_ACCOUNT_CODE` at offset `i`, until `i==code_length`.
+load_code_loop:
+    // stack: i, code_length, codehash, retdest
+    DUP2 DUP2 EQ
+    // stack: i == code_length, i, code_length, codehash, retdest
+    %jumpi(load_code_check)
+    PROVER_INPUT(account_code::get)
+    // stack: opcode, i, code_length, codehash, retdest
+    DUP2
+    // stack: i, opcode, i, code_length, codehash, retdest
+    %mstore_current(@SEGMENT_KERNEL_ACCOUNT_CODE)
+    // stack: i, code_length, codehash, retdest
+    %increment
+    // stack: i+1, code_length, codehash, retdest
+    %jump(load_code_loop)
+
+// Check that the hash of the loaded code equals `codehash`.
+load_code_check:
+    // stack: i, code_length, codehash, retdest
+    POP
+    // stack: code_length, codehash, retdest
+    %stack (code_length, codehash, retdest) -> (@SEGMENT_KERNEL_ACCOUNT_CODE, 0, code_length, codehash, retdest, code_length)
+    GET_CONTEXT
```

Comment:
nit: I'd suggest using context=0 for a couple very minor reasons
- Seems slightly convenient since we can include 0 in `%stack`
- Arguably it might seem more conventional for the kernel to use "its own" memory, rather than memory "owned by" a userspace call. (Admittedly we already violate this guideline with `ContextMetadata`, but most other kernel stuff uses context=0.)

Reply:
Good points, I changed it